include/LightGBM/dataset.h

```diff
@@ -212,6 +262,16 @@ class Metadata {
   */
   inline int64_t num_init_score() const { return num_init_score_; }
 
+  /*!
+  * \brief Get number of classes
+  */
+  inline int32_t num_classes() const {
+    if (num_data_) {
+      return static_cast<int>(num_init_score_ / num_data_);
```

Comment:
is `num_init_score_` always set to non zero values?

Reply:
no, sometimes there are no initial scores.  Should we return 1 even in the case of no initial score inputs?

Reply:
but when there is no num_init_score_, num_classes could be > 1.

Reply:
Ok, makes sense to return 1 for all non-multiclass scenarios, so I made the change.

Reply:
@svotaw is this addressed?

Reply:
it looks like I missed this one.  Is there a particular concern/bug?  What is it that you suggest it should return?  This is currently used as part of allocation.  If num_init_score_ == 0, this is not used.  Happy to fix it if needed.

Reply:
`num_init_score_` does not always have values. 
I believe the `num_class` is a hyper-parameter, which can get from config. So I am not sure why you use `num_init_score_` to get `num_class`. 

If `dataset->num_classes()` is currently only used for init_score assignment, maybe we should use a different name to avoid future bugs.

Reply:
makes sense.  How about num_init_score_classes()?

Reply:
it looks good to me